prompting/storage/storage.go

```diff
@@ -105,7 +108,7 @@ func (pd *PromptsDB) Set(req *notifier.Request, allow bool, extras map[string]st
 	if !osutil.IsDirectory(path) {
 		path = filepath.Dir(path)
 	}
-	if findPathInSubdirs(allowWithSubdirs, path) {
+	if alreadyAllowed, err := findPathInSubdirs(allowWithSubdirs, path); err == nil && !(alreadyAllowed ^ allow) {
```

Comment:
(nitpick) not a concern right now but if we ever return more/different errors from findPathInSubdirs() than ErrNoSavedDecision (which is admittedly unlikely) this would not handle real errors. So maybe change to something like this (maybe buggy but hopefully you get the idea)
(nipck^2) given that both alreadyAllowed and allow are bool could the bitwise not XOR be simplified somehow? 
```go
	alreadyAllowed, err := findPathInSubdirs(allowWithSubdirs, path); 
        if err != nil && err != ErrNoSavedDecision {
            return err
         }
         if err == nil && (reladyAllow != allow) {
             return nil
        }
```

Reply:
Good catch, this is a really great suggestion. I implemented your change, with the one tweak that it should have `(alreadyAllow == allow)`, since if it's a subdir and we're setting the same permissions, we should do nothing (`return nil`).

Open question: If we already had, say, allow `true` for `/home/test/Documents` but tried to call `Set()` with allow `true` for `/home/test`, `Set()` should presumably add it to the database, since it is more broad than the previous mapping, but should it remove the more specific `/home/test/Documents` allowance from the database as well since it is now redundant?

Reply:
Thanks for the update! About the open question - let's just leave a TODO for now, I am not sure yet how final this "database" is anyway.

Reply:
Sounds good. I added a larger comment about it in the apparmor prompting notes document, so we can discuss later if desired.